### PR TITLE
Benchmarks: Add Benchmark - Add the source code of rocm kernel launch overhead benchmark. 

### PR DESCRIPTION
**Pull request by yukirora**
**Description**
Add the source code of rocm kernel launch overhead benchmark. 

**Major Revision**
- Revise cmake build logic to support both cuda and rocm

### CODE REVIEW

File: superbench/benchmarks/micro_benchmarks/kernel_launch_overhead/CMakeLists.txt [hunk 1]
```diff
@@ -2,10 +2,57 @@
 # Licensed under the MIT License.
 
 cmake_minimum_required(VERSION 3.18)
-project(kernel_launch_overhead LANGUAGES CUDA CXX)
 
-include(../cuda_common.cmake)
+project(kernel_launch_overhead LANGUAGES CXX)
```
Comment (cp5555) [inline superbench/benchmarks/micro_benchmarks/kernel_launch_overhead/CMakeLists.txt:6]: In my mind, we should a separate PR which superbench should have cuda_common.cmake and rocm_common.cmake. For every benchmark, it should call both cmake directly.

File: superbench/benchmarks/micro_benchmarks/kernel_launch_overhead/CMakeLists.txt [hunk 2]
```diff
@@ -2,10 +2,57 @@
...
+    set(HIP_PATH ${ROCM_PATH}/hip)
+    else()
+    set(HIP_PATH $ENV{HIP_PATH})
+    endif()
+    # Check if ROCm environment
+    if(EXISTS ${HIP_PATH})
+        # Search for hip in common locations
+        list(APPEND CMAKE_PREFIX_PATH ${HIP_PATH} ${ROCM_PATH})
+        set(CMAKE_MODULE_PATH "${HIP_PATH}/cmake" ${CMAKE_MODULE_PATH})
+        set(CMAKE_MODULE_PATH "${HIP_PATH}/lib/cmake/hip" ${CMAKE_MODULE_PATH})
```
Comment (guoshzhao) [inline superbench/benchmarks/micro_benchmarks/kernel_launch_overhead/CMakeLists.txt:38]: As Peng's comment, could we put ENV part into separate file for common usage?

Comment (yukirora) [reply superbench/benchmarks/micro_benchmarks/kernel_launch_overhead/CMakeLists.txt]: Sure

File: superbench/benchmarks/micro_benchmarks/kernel_launch_overhead/CMakeLists.txt (hunk 1 above)
Comment (yukirora) [reply superbench/benchmarks/micro_benchmarks/kernel_launch_overhead/CMakeLists.txt:6]: Thanks for your suggestion, I have submitted another pr.

**Review by cp5555 — APPROVED** [review]

**Review by cp5555 — APPROVED** [review]

File: superbench/benchmarks/micro_benchmarks/kernel_launch_overhead/CMakeLists.txt [hunk 3]
```diff
@@ -2,10 +2,39 @@
...
+    add_executable(kernel_launch_overhead kernel_launch.cu)
+    set_property(TARGET kernel_launch_overhead PROPERTY CUDA_ARCHITECTURES ${NVCC_ARCHS_SUPPORTED})
+    install(TARGETS kernel_launch_overhead RUNTIME DESTINATION bin)
+
+# ROCm environment
+elseif(HIP_FOUND)
+    message(STATUS "Found HIP: " ${HIP_VERSION})
+    
+    # Convert cuda code to hip code inplace
+    execute_process(COMMAND hipify-perl -inplace -print-stats kernel_launch.cu
```
Comment (guoshzhao) [inline superbench/benchmarks/micro_benchmarks/kernel_launch_overhead/CMakeLists.txt:27]: will hipify-perl always exist?

Comment (yukirora) [reply superbench/benchmarks/micro_benchmarks/kernel_launch_overhead/CMakeLists.txt:27]: yes, in amd machines, it's build-in

**Review by guoshzhao — APPROVED** [review]

**Pull request merged**